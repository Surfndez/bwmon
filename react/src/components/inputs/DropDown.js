import React from 'react';
import { Select, MenuItem } from '@material-ui/core';

const DropDown = ({ items, ...props }) => {
    return items && props.value ? (
        <Select {...props}>
            {items.map((item, index) => (
                <MenuItem key={index} value={item}>
                    {item}
                </MenuItem>
            ))}
        </Select>
    ) : null;
};

export default DropDown;
