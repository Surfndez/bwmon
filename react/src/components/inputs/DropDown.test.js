import React from 'react';
import { create } from 'react-test-renderer';
import DropDown from './DropDown';

describe('DropDown', () => {
    const handleChange = (event) => {
        console.log('handle change');
    };

    test('renders correctly with items', () => {
        const tree = create(<DropDown id="12345" value="ZYXUW" onChange={handleChange} items={['ABC', 'ZYXUW', 'WGKF']} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders correctly with no items', () => {
        const tree = create(<DropDown />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('renders correctly with no value', () => {
        const tree = create(<DropDown items={['ABC', 'ZYXUW', 'WGKF']} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
