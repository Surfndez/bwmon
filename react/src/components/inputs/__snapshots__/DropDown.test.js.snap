// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropDown renders correctly with items 1`] = `
<div
  className="MuiInputBase-root MuiInput-root MuiInput-underline"
  onClick={[Function]}
>
  <div
    aria-haspopup="listbox"
    aria-labelledby="12345"
    className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
    id="12345"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    role="button"
    tabIndex={0}
  >
    ZYXUW
  </div>
  <input
    aria-hidden={true}
    className="MuiSelect-nativeInput"
    onAnimationStart={[Function]}
    onChange={[Function]}
    tabIndex={-1}
    value="ZYXUW"
  />
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root MuiSelect-icon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M7 10l5 5 5-5z"
    />
  </svg>
</div>
`;

exports[`DropDown renders correctly with no items 1`] = `null`;

exports[`DropDown renders correctly with no value 1`] = `null`;
